packages/remix-dev/compiler/plugins/cssFilePlugin.ts

```diff
@@ -15,31 +18,35 @@ function normalizePathSlashes(p: string) {
  * This plugin loads css files with the "css" loader (bundles and moves assets to assets directory)
  * and exports the url of the css file as its default export.
  */
-export function cssFilePlugin(options: {
-  mode: CompileOptions["mode"];
-  rootDirectory: string;
+export function cssFilePlugin({
+  config,
+  options,
+}: {
+  config: RemixConfig;
+  options: CompileOptions;
 }): esbuild.Plugin {
   return {
     name: "css-file",
 
     async setup(build) {
       let buildOps = build.initialOptions;
 
+      let postcssProcessor = await getPostcssProcessor({ config });
+
       build.onLoad({ filter: /\.css$/ }, async (args) => {
-        let { outfile, outdir, assetNames } = buildOps;
         let { metafile, outputFiles, warnings, errors } = await esbuild.build({
           ...buildOps,
           minify: options.mode === "production",
           minifySyntax: true,
           metafile: true,
           write: false,
-          sourcemap: false,
+          sourcemap: Boolean(options.sourcemap && postcssProcessor), // We only need source maps if we're processing the CSS with PostCSS
           incremental: false,
           splitting: false,
           stdin: undefined,
           outfile: undefined,
-          outdir: outfile ? path.dirname(outfile) : outdir,
-          entryNames: assetNames,
+          outdir: config.assetsBuildDirectory,
```

Comment:
This was needed so the build is consistent between browser and server builds. With the original code, the server and client get different source maps which results in multiple, differing copies of the same styles being generated for a single Remix build. This is because the hash of the source map is in the generated CSS file since it contains a comment with the file name. (This was a tough one to track down!)